lib/internal/crypto/keys.js

```diff
@@ -329,6 +331,21 @@ function createPublicKey(key) {
 }
 
 function createPrivateKey(key) {
+  if (typeof key === 'object' && key &&
+      (key.type === 'poly1305' || key.type === 'siphash')) {
+    const secret = key.key;
+
+    if (!isArrayBufferView(secret)) {
+      const allowed = ['Buffer', 'TypedArray', 'DataView'];
+      throw new ERR_INVALID_ARG_TYPE('key', allowed, secret);
```

Comment:
```suggestion
      throw new ERR_INVALID_ARG_TYPE('key.key', allowed, secret);
```
??

Reply:
I initially wrote something like what you suggest but then I noticed similar code a few lines up and decided to go with that for consistency.

Reply:
Sigh. Yeah, there's quite a bit of inconsistency in the code base for this case. `ERR_INVALID_ARG_TYPE` uses different logic for generating the message string for when reporting directly on the argument or a property on the argument. In this case, since it is `key.key` that is being tested, `key.key` should be used in the error message and the other similar cases likely should be fixed.